code/modules/worldgen/GenerateMining.dm

```diff
@@ -319,6 +319,16 @@ var/list/miningModifiersUsed = list()//Assoc list, type:times used
 	game_start_countdown?.update_status("Setting up mining level...\nGenerating terrain...")
 	miningZ = D.generate(miningZ)
 
+	// remove temporary areas
+	for (var/turf/T in get_area_turfs(/area/noGenerate))
```

Comment:
those areas should delete themselves already tho?

Reply:
They don't, and there's a noticeable difference in lighting, especially on trench-mode z5.

Reply:
Hmm, they should delete themselves in their New() instead of this

Reply:
If they did, that would mean the mining z-level wouldn't generate anything. The mining z-level generates during the loading screen, so the game's logic is active during that time.

Reply:
after a sleep?

Reply:
There are other places this logic could be put, but it would be much messier to have it done by the area or the turf as neither one would know when generation actually finishes.